tensorflow/contrib/kinesis/kernels/kinesis_dataset_ops.cc

```diff
@@ -0,0 +1,368 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <aws/core/Aws.h>
+#include <aws/core/config/AWSProfileConfigLoader.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/kinesis/KinesisClient.h>
+#include <aws/kinesis/model/DescribeStreamRequest.h>
+#include <aws/kinesis/model/GetRecordsRequest.h>
+#include <aws/kinesis/model/GetShardIteratorRequest.h>
+#include <aws/kinesis/model/PutRecordsRequest.h>
+#include <aws/kinesis/model/ShardIteratorType.h>
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/platform/s3/aws_crypto.h"
+
+namespace tensorflow {
+namespace {
+
+Aws::Client::ClientConfiguration& GetDefaultClientConfig() {
+  static mutex mu(LINKER_INITIALIZED);
+  static bool init(false);
+  static Aws::Client::ClientConfiguration config;
+
+  std::lock_guard<mutex> lock(mu);
+
+  if (!init) {
+    const char* endpoint = getenv("KINESIS_ENDPOINT");
+    if (endpoint) {
+      config.endpointOverride = Aws::String(endpoint);
+    }
+    const char* region = getenv("AWS_REGION");
+    if (region) {
+      config.region = Aws::String(region);
+    } else {
+      // Load config file (e.g., ~/.aws/config) only if AWS_SDK_LOAD_CONFIG
+      // is set with a truthy value.
+      const char* load_config_env = getenv("AWS_SDK_LOAD_CONFIG");
+      string load_config =
+          load_config_env ? str_util::Lowercase(load_config_env) : "";
+      if (load_config == "true" || load_config == "1") {
+        Aws::String config_file;
+        // If AWS_CONFIG_FILE is set then use it, otherwise use ~/.aws/config.
+        const char* config_file_env = getenv("AWS_CONFIG_FILE");
+        if (config_file_env) {
+          config_file = config_file_env;
+        } else {
+          const char* home_env = getenv("HOME");
+          if (home_env) {
+            config_file = home_env;
+            config_file += "/.aws/config";
+          }
+        }
+        Aws::Config::AWSConfigFileProfileConfigLoader loader(config_file);
+        loader.Load();
```

Comment:
What happens if this returns `false`? Can you add add something to report an error back to the user?

Reply:
@mrry I updated the PR. Now unsuccessful `Load()` will generate a warning, and the following operations will be skipped. This is similar to:
https://github.com/aws/aws-sdk-cpp/blob/970a729a7d3541b2fd7948f413992e8a851ec329/aws-cpp-sdk-core/source/config/AWSProfileConfigLoader.cpp